src/Tools/dotnet-monitor/Startup.cs

```diff
@@ -173,5 +179,27 @@ public void ConfigureServices(IServiceCollection services)
                 builder.MapControllers();
             });
         }
+
+        private static void LogBoundAddresses(IFeatureCollection features, AddressListenResults results, ILogger logger)
+        {
+            IServerAddressesFeature serverAddresses = features.Get<IServerAddressesFeature>();
```

Comment:
Is there a reason we fall back to the IServerAddressFeature instead of simply passing the urls along to the AddressListenResults if they succeed in binding, and having that structure contain two separate collections, one for default and one for metric urls?

Reply:
We aren't actually binding the addresses, we are setting up the options for each address when calling KestrelServerOptions.Listen*. Binding happens latter when the Kestrel server starts up. So the urls that we have while setting up the listening options still have 0 for their ports if 0 was specified on the command line.